anoma/namada-interface
# feat/374 - Hook up IBC to Approval and Ledger flows

Resolves #374 

_Note 1_: As a part of this PR to update the Bridge page for IBC transfers, I did a bit of refactoring to remove the `transfers` slice, as that part of the state was going completely unused.

_Note 2_: The submission of IBC is currently disabled at the end of the submission process, so we can follow through the approval flow, but `v0.20.1` has some issues (will crash for some reason), so this will be best enabled and tested in `0.22.x`

_Note 3_: I left Bridge transfer unchanged (it is currently not properly hooked up) to focus on IBC

- [x] IBC Transfers follow the Approvals flow and can be submitted (_Note_ Without a proper IBC set up, i.e., with multiple chains and a relayer running, these tx will not be applied)
- [x] IBC transfers work with Ledger approvals
  - Tx submission is disabled for Ledger, but the signing process should work as a _Custom_ tx (though the Ledger is unable to parse information from it currently)

### Testing

This assumes you have the Keplr extension installed!

- Create a mnemonic account in extension, or connect to Ledger
- Load funds into account via faucet
- Navigate to the `Bridge` tab
- Select the account+token that has a balance
- Click to add a channel ID (e.g., `channel-0`)
- Enter recipient, or you can also click `Load Accounts from Keplr extension` and it will load the recipient address
- Specify a valid amount, then click `Submit` - You should be prompted to approve (and authenticate or unlock Ledger). Note that currently approving will not actually submit the tx! You should instead get a "Success" notification. Ledger approvals will need to also approve on the device (it is currently parsed as a `Custom` tx, so you won't see any details on the device, as it cannot parse any details)
